src/world_new/Ball.cpp

```diff
@@ -114,12 +114,12 @@ void Ball::updateBallAtRobotPosition() noexcept {
 
     auto world = World::instance()->getWorld();
 
-    auto robotWithBall = world->whichRobotHasBall();
-    if (!robotWithBall) {
+    std::optional<rtt::world_new::view::RobotView> robotWithBall = world->whichRobotHasBall();
+    if (!robotWithBall.has_value()) {
         return;
     }
 
-    std::pair<uint8_t, Team> newRobotIdTeam = {robotWithBall->getId(), robotWithBall->getTeam()};
+    std::pair<uint8_t, Team> newRobotIdTeam = {(*robotWithBall)->getId(), (*robotWithBall)->getTeam()};
```

Comment:
Was this change what fixed a bug? I know this caused UB before as well...

Reply:
No, this does not fix the bug. This just now returns a RobotView instead of a RobotPtr, as it should

Reply:
Wait so the extra dereference means it becomes a view?

Reply:
world->whichRobotHasBall() returns an std::optional<RobotView>, because there might not be a robot that has the ball. (*robot) then unpacks that std::optional, which leaves you with a RobotView.

Reply:
Ahh okay. I didn't know that